mock-data/generateData.js

```diff
@@ -0,0 +1,111 @@
+const Chance = require('chance');
+const Moment = require('moment');
+const MomentRange = require('moment-range');
+
+const chance = new Chance();
+const moment = MomentRange.extendMoment(Moment);
+
+const LISTINGS_DEFAULT = 100;
+const OWNERS_DEFAULT = 100;
+const USERS_DEFAULT = 755;
+const MIN_STAY_DEFAULT = 2;
+const MAX_STAY_DEFAULT = 11;
+
+const createBooking = (startDate, duration, listingID, users) => {
+  const start = startDate.format('YYYY-MM-DD');
+  const end = startDate.clone().add(duration, 'days').format('YYYY-MM-DD');
+  const userID = chance.integer({ min: 1, max: users });
+  return [listingID, userID, start, end];
+};
+
+const addMockBookings = (listingID, momentRange, users) => {
+  const mockBookings = [];
+  const popularity = chance.integer({ min: 2, max: 8 });
+  let available = true;
+  let prevBookingDuration;
+  let count = 0;
+  let daysAway = 0;
+  let reduced = 0;
+
+  for (const day of momentRange.by('day')) {
+    count = available ? count : count += 1;
+    if (!available && count === prevBookingDuration) {
+      available = true;
+      count = 0;
+    }
+    if (available && popularity >= chance.integer({ min: 1, max: 10 + reduced })) {
+      available = false;
+      prevBookingDuration = chance.integer({ min: MIN_STAY_DEFAULT, max: MAX_STAY_DEFAULT });
+      mockBookings.push(createBooking(day, prevBookingDuration, listingID, users));
+    }
+    daysAway += 1;
+    reduced = Math.floor(daysAway * 0.25);
+  }
+  return mockBookings;
+};
+
+const generateAllBookings = (listings = LISTINGS_DEFAULT, users = USERS_DEFAULT) => {
+  const today = moment().format('YYYY-MM-DD');
+  const periodEnd = moment(today).clone().add(2, 'years').format('YYYY-MM-DD');
+  const bookingRange = moment.range(today, periodEnd);
+  const allListingBookings = [];
+
+  for (let i = 1; i <= listings; i += 1) {
+    const listingID = i;
```

Comment:
Shouldn't listingID be coming from foreign key, not created by you?

Reply:
I'm still a little iffy on how to explicitly set a foreign key for a particular row, and not just the whole column tbh, I watched a few videos where they set up the references/then assigned the foreign key to a particular table ID when inserting data (mysql complained when I accidentally generated too few users, and tried to assign keys that did not exist, but it didn't have an issue assigning them to existing keys), anyways, if you have any thoughts/insights I'd definitely appreciate it, thanks again man!